### ISSUE TITLE
valid-expect does not work with then

### ISSUE THREAD

**Issue by glickel** [jest-community/eslint-plugin-jest#347]
Hello, I have a problem with `jest/valid-expect` rule, it seems to not work when I put a `.then` after a `.rejects` or a `.resolves` like this:

```
return expect(functionReturningAPromise()).resolves.toEqual(1).then(() => {
    expect(someMock).toHaveBeenCalledTimes(1);
});
```

When I disable the rule `jest/valid-expect` or I remove the `.then` it works well.

**Comment by SimenB** [comment]
I'd recommend using `async-await` instead

```js
await expect(functionReturningAPromise()).resolves.toEqual(1)
expect(someMock).toHaveBeenCalledTimes(1);
```

However, the code you write _is_ valid, so it shouldn't complain

/cc @yatki

**Comment by yatki** [comment]
@SimenB #352 hopefully this would fix it.

**Comment by glickel** [comment]
Thank you :)
